lightbug_http/sys/client.mojo

```diff
@@ -16,26 +16,27 @@ from lightbug_http.http import HTTPRequest, HTTPResponse, encode, split_http_str
 from lightbug_http.header import ResponseHeader
 from lightbug_http.sys.net import create_connection
 from lightbug_http.io.bytes import Bytes
+from lightbug_http.uri import URI
 
 
 struct MojoClient(Client):
-    var fd: c_int
     var host: StringLiteral
     var port: Int
     var name: String
 
     fn __init__(inout self) raises:
-        self.fd = socket(AF_INET, SOCK_STREAM, 0)
         self.host = "127.0.0.1"
         self.port = 8888
         self.name = "lightbug_http_client"
 
     fn __init__(inout self, host: StringLiteral, port: Int) raises:
-        self.fd = socket(AF_INET, SOCK_STREAM, 0)
         self.host = host
         self.port = port
         self.name = "lightbug_http_client"
 
+    fn get_socket(self) -> c_int:
```

Comment:
I noticed reusing the socket fd causes subsequent requests to fail, I'm not sure if there's a way to make it work but from my understanding you aren't really supposed to in this case?

Reply:
I think this is because it was trying to open a new connection on the same file descriptor.
Just pushed a commit that stores the connection on the `MojoClient` struct and then reuses it for the redirect, I think this is best practice for HTTP/1.1. The only thing is that if the redirect is pointing to a different host then we should close the original connection and open a new one, didn't implement that yet. 
Could you take a look at the changes and let me know what you think?

Reply:
@bgreni I think in the latest version the conn is not stored on the client struct anymore, were you able to work around this some other way?

Reply:
For now I'm just shutting down the connections. Correct me if I'm wrong, but in order to properly implement persistent connections we need to be a bit more intelligent about since the various requests being sent aren't necessarily being sent to the same host? So we may need to manage multiple connections at a time?

Reply:
Yes, true, if the host is different a new connection has to be established. Alright, let's leave this for the future then!